aiounifi/interfaces/connectivity.py

```diff
@@ -74,16 +74,13 @@ async def login(self) -> None:
                 LOGGER.error("Login failed '%s'", data)
                 raise ERRORS.get(data["meta"]["msg"], AiounifiException)
 
-        elif response.content_type == "text/html":
+            if (csrf_token := response.headers.get("x-csrf-token")) is not None:
+                self.headers["x-csrf-token"] = csrf_token
+
+        else:
             LOGGER.debug("Login Failed not JSON: '%s'", bytes_data)
             raise RequestError("Login Failed: Host starting up")
 
```

Comment:
This can still be done after the if/else
```suggestion
        else:
            LOGGER.debug("Login Failed not JSON: '%s'", bytes_data)
            raise RequestError("Login Failed: Host starting up")

        if (csrf_token := response.headers.get("x-csrf-token")) is not None:
            self.headers["x-csrf-token"] = csrf_token
```

Reply:
I suppose it could - it was broken with the tests having `text/json` - If we move it out do we need to put the `response.status == HTTPStatus.OK` test back in line 81?  I think so....

Reply:
I'm not sure. I would have to check the commit history

Reply:
The code looks quite a bit different today from when it was added, so I think its okay without the check

https://github.com/Kane610/aiounifi/pull/140/files

Reply:
Im gonna merge this so I can get out a fix for 2024.4.4.

Reply:
I think it's ok either way - at least it knows when a login failed :)
Thanks for all your work on this "fun" stuff.